src/chapters/guacamole-rest-api.xml

```diff
@@ -0,0 +1,836 @@
+        also beneficial to understand how the API works. This allows you to have the
+        flexibility of incorporate Guacamole within your application
+        without having to worry about which language to use or what packages
+        to download.
+    </para>
+
+
+    <section xml:id="base_url">
+        <title>Base URL</title>
+        <para>There are four things you will need to set the base url before
+        you can start talking to the endpoints. </para>
+
+        <itemizedlist>
+            <listitem>
+                <para>Method</para>
+                <emphasis>
+                Protocol your Guacamole instance is using e.g. http, https
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>Hostname</para>
+                <emphasis>
+                Where your Guacamole is hosted e.g. guacamole.example.com
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>URL Path</para>
+                <emphasis>
+                The pathway to your Guacamole (where you can see the login page) e.g /guacamole
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>Data Source</para>
+                <emphasis>
+                Database your Guacamole is running on e.g. mysql, postgresql
+                </emphasis>
+            </listitem>
+        </itemizedlist>
+
+        <para>When you have the data listed above you can set your base url which will look
+            like the url below and you can start talking to the API. </para>
+
+        <informalexample>
+        <programlisting><replaceable>http://</replaceable> <replaceable>guacamole.example.com</replaceable> <replaceable>/guacamole</replaceable> /api/data/ <replaceable>mysql</replaceable></programlisting>
+        </informalexample>
+
+    </section>
+
+    <section xml:id="authentication">
+        <title>Authentication</title>
+        <para>You will need to request an authentication token before being able to
+        call any endpoint.</para>
+
+        <para>You can request your tokens from the endpoint below:
+        <informalexample><programlisting>/api/tokens</programlisting></informalexample>
+        </para>
+
+        <para>Include your username and password in the form/body of the request.</para>
+        <informalexample>
+<programlisting>{
+    "username": <replaceable>"adminuser"</replaceable>,
+    "password": <replaceable>"adminpassword"</replaceable>
+}</programlisting>
+        </informalexample>
+
+        <para>You will recieve the following authentication object: </para>
+
+        <informalexample>
+<programlisting>{
+    authToken: <replaceable>{token}</replaceable>,
+    username: <replaceable>'adminuser'</replaceable>,
+    dataSource: <replaceable>'mysql'</replaceable>,
+    availableDataSources: <replaceable>[ 'mysql', 'mysql-shared' ]</replaceable>
+}</programlisting>
+        </informalexample>
```

Comment:
"will" may misleading here when the example is specific to the response that would be seen if the MySQL backend is being used. If the example is specific to a particular case, that should probably be noted.